qa/tasks/cephfs/test_admin.py

```diff
@@ -2154,3 +2186,163 @@ def test_fs_authorize(self):
                 args=(f'fs authorize {self.fs.name} {self.CLIENT_NAME} / '
                       f'{wrong_perm}'), retval=self.EXPECTED_ERRNO,
                 errmsgs=self.EXPECTED_ERRMSG)
+
+
+class TestFSFail(TestAdminCommands):
+
+    MDSS_REQUIRED = 2
+    CLIENTS_REQUIRED = 1
+
+    def test_with_health_warn_oversize_cache(self):
+        '''
+        Test that, when health warning MDS_CACHE_OVERSIZE is present for an
+        MDS, command "ceph fs fail" fails without confirmation flag and passes
+        when confirmation flag is passed.
+        '''
+        health_warn = 'MDS_CACHE_OVERSIZED'
+        self.config_set('mds', 'mds_cache_memory_limit', '1K')
+        self.config_set('mds', 'mds_health_cache_threshold', '1.00000')
+        active_mds_id = self.fs.get_active_names()[0]
+
+        self.mount_a.open_n_background(".", 400)
+        self.wait_till_health_warn(health_warn, active_mds_id)
+
+        # actual testing begins now.
+        errmsg = 'mds_cache_oversized'
+        self.negtest_ceph_cmd(args=f'fs fail {self.fs.name}',
+                              retval=1, errmsgs=errmsg)
+        self.run_ceph_cmd(f'fs fail {self.fs.name} --yes-i-really-mean-it')
```

Comment:
Should we also test it without the `--yes-i-really-mean-it` option, which the failure is expected ? And the same for the others.

Reply:
> Should we also test it without the --yes-i-really-mean-it

We're doing that in the line just above this one - `self.negtest_ceph_cmd()`.

Reply:
Could you point me which line ?

Reply:
Sure - https://github.com/ceph/ceph/pull/56066/files#diff-534e44897921b6dcac93d2c7bb1cf04a3727bf1f025c579f83d898e2c2257400R2212-R2213

Reply:
Oh, okay, just miss reading that line, I was finding the `run_ceph_cmd()` instead.